spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/aggregation/ProjectionOperation.java

```diff
@@ -1700,4 +1711,35 @@ public Document toDocument(AggregationOperationContext context) {
 			return new Document(field.getName(), expression.toDocument(context));
 		}
 	}
+
+	/**
+	 * A {@link Projection} including all top level fields of the given target type mapped to include potentially
+	 * deviating field names.
+	 *
+	 * @since 2.2
+	 * @author Christoph Strobl
+	 */
+	static class TypeProjection extends Projection {
+
+		private final Class<?> type;
+
+		TypeProjection(Class<?> type) {
+
+			super(Fields.field(type.getSimpleName()));
+			this.type = type;
+		}
+
+		/*
+		 * (non-Javadoc)
+		 * @see org.springframework.data.mongodb.core.aggregation.ProjectionOperation.Projection#toDocument(org.springframework.data.mongodb.core.aggregation.AggregationOperationContext)
+		 */
+		@Override
+		public Document toDocument(AggregationOperationContext context) {
+
+			Document projections = new Document();
+			ReflectionUtils.doWithFields(type, it -> projections.append(it.getName(), 1));
```

Comment:
We probably should differentiate here and rely on our meta-model. One might want to use property accessors instead or hand in a projection interface.